bootcamp_project/Module 8/More_OOP_Concepts/VideoGame_Store_OOP_EMira1.ts

```diff
@@ -0,0 +1,79 @@
+class VideoGame {
+
+    // Static attributes
+    static availableGames: VideoGame[] = [];
+    static soldOutGames: VideoGame[] = [];
+    static totalMoneyEarned: number;
+
+
+    // Instance attributes
+    title: string;
+    developer: string;
+    availableCopies: number;
+    price: number;
+    copiesSold: number;
+
+    constructor(title: string, developer: string, availableCopies: number, price: number, copiesSold: number) {
+        this.title = title;
+        this.developer = developer;
+        this.availableCopies = availableCopies;
+        this.price = price;
+        this.copiesSold = copiesSold;
+
+        VideoGame.availableGames.push(this);
+    }
+
+    // Instance methods
+    isAvailable(): boolean {
+    return this.availableCopies > 0
+    }
+
+    sell(numberOfCopies:number = 1): number {
+
+        try {
+            if ( this.isAvailable() ) {
+                this.availableCopies -= numberOfCopies;
+                VideoGame.totalMoneyEarned += numberOfCopies;
```

Comment:
Since this was changed to 'total money earned', I think this should be 
```suggestion
                VideoGame.totalMoneyEarned += this.price*numberOfCopies;
```